llvm/lib/Target/WebAssembly/WebAssemblyISelLowering.cpp

```diff
@@ -3210,6 +3213,56 @@ static SDValue performTruncateCombine(SDNode *N,
   return truncateVectorWithNARROW(OutVT, In, DL, DAG);
 }
 
+static SDValue performVSelectCombine(SDNode *N,
+                                     TargetLowering::DAGCombinerInfo &DCI) {
+  // In the tablegen.td, vselect A B C -> bitselect B C A
+
+  // SCENARIO 1
+  // vselect Y, <0>, X
```

Comment:
We need to be careful here because selects block poison in either operands, so if remove the select then we need to freeze the arguments